src/test/regress/sql/functional_deps.sql

```diff
@@ -257,3 +260,145 @@ EXECUTE foo;
 ALTER TABLE articles DROP CONSTRAINT articles_pkey RESTRICT;
 
 EXECUTE foo;  -- fail
+
+-- Known issue - in case there is a column in the target list that is not
+-- explicitly listed in the GROUP BY clause but has a functional dependency on
+-- another column in the GROUP BY clause, Postgres planner might output column
+-- values in grouped rows, that actually cannot be uniquely identified. As it
+-- can introduce instability in the tests, all the queries below are
+-- temporarily tested only with GPORCA.
+-- TODO: remove the hard-coded optimizer option below once this issue is fixed.
+SET optimizer = on;
+
+-- start_ignore
+DROP TABLE IF EXISTS test_table1;
+DROP TABLE IF EXISTS test_table2;
+DROP TABLE IF EXISTS test_table3;
+-- end_ignore
+
+CREATE TABLE test_table1 (a int PRIMARY KEY, b text, c int);
+INSERT INTO test_table1 VALUES 
+(1, 'text1', 0),
+(2, 'text2', 1),
+(3, 'text3', 0),
+(4, 'text4', 1),
+(5, 'text0', 0),
+(6, 'text1', 1),
+(7, 'text2', 0),
+(8, 'text3', 1),
+(9, 'text4', 0),
+(10, 'text0', 1),
+(11, 'text1', 0),
+(12, 'text2', 1),
+(13, 'text3', 0),
+(14, 'text4', 1),
+(15, 'text0', 0),
+(16, 'text1', 1),
+(17, 'text2', 0),
+(18, 'text3', 1),
+(19, 'text4', 0),
+(20, 'text0', 1),
+(21, 'text1', 0),
+(22, 'text2', 1),
+(23, 'text3', 0),
+(24, 'text4', 1),
+(25, 'text0', 0);
+
+CREATE TABLE test_table2 (a int, b text, c int, CONSTRAINT id_t2 PRIMARY KEY(a,c) );
+INSERT INTO test_table2 VALUES 
+(1, 'text|1|-1|', -1),
+(1, 'text|1|1|',   1),
+(2, 'text|2|-1|', -1),
+(2, 'text|2|1|',   1),
+(3, 'text|3|-1|', -1),
+(3, 'text|3|1|',   1),
+(4, 'text|4|-1|', -1),
+(4, 'text|4|1|',   1),
+(5, 'text|5|-1|', -1),
+(5, 'text|5|1|',   1);
+
+CREATE TABLE test_table3 (a int PRIMARY KEY, b text, c int);
+INSERT INTO test_table3 VALUES 
+(1, 't3-text1', 0),
+(2, 't3-text2', 1),
+(3, 't3-text3', 0),
+(4, 't3-text4', 1),
+(5, 't3-text0', 0),
+(6, 't3-text1', 1),
+(7, 't3-text2', 0),
+(8, 't3-text3', 1),
+(9, 't3-text4', 0),
+(10, 't3-text0', 1);
+
+-- Check simple group by clause
+SELECT a, b FROM test_table1 GROUP BY a ORDER BY a;
```

Comment:
Why in all your new tests do you only check the execution result and not the plan?

Reply:
Because the aim of the added tests is to verify that ORCA can provide valid plans for them.
For that purpose checking the output and absence of fallback messages is enough.

For some cases the plan could be not optimal, and, thus, I would like not to place checks of plans in this test.
Optimization of plans provided by ORCA is outside the scope of this task.

Reply:
> But in the tests above, both the plan and the results are checked!

Tests above are old tests that were used for testing postgres planner. Originally I disabled them for GPORCA, but per your comment (refer to https://github.com/arenadata/gpdb/pull/746#pullrequestreview-1881709553 ) the output was split into 2 files (original and _optimizer.out). And all tests were enabled for ORCA as well to minimize delta in the test.
 
Adding checking of plans into the tests without intention to optimize ORCA's planning procedure doesn't make much sence.

> Additionally, if you only want to test ORCA, you can try adding unit (mdp) tests.

All changes apply before translation to DXL, so unit test based on minidumps are not applicable here.

Reply:
> Adding checking of plans into the tests without intention to optimize ORCA's planning procedure doesn't make much sence.

Why? This way we will test not only the results, but also the plans and will immediately see if they change.

Reply:
> Why? This way we will test not only the results, but also the plans and will immediately see if they change.

To conduct a proper functional test, the output of the test should be determined by the specification of the functionality.
In case we examine the output of the query, it is straightforward and unambiguous. But speaking about the plans of these queries - we do not have unambiguous expectations or specification how they should look like (at least I'm not aware about such specs). What you propose is to just launch explain on the new queries in the test, and copy/paste the output to the expected answers, right?

So, every time somebody changes ORCAs logic, it may trigger a test failure. And every time he will need to waste time investigating whether it's his fault or the test should be updated just because his changes introduce some optimization.

Arguing that such things were done for the original tests would not be correct - as I've said, my original intention was to disable them for GPORCA completely, and leave this legacy only for postgres planner.
But I compromised to enable them during the review. And I strongly want to minimize such cases.

Reply:
> So, every time somebody changes ORCAs logic, it may trigger a test failure. And every time he will need to waste time investigating whether it's his fault or the test should be updated just because his changes introduce some optimization.

Is this bad? This will mean that more code is covered by tests!

Reply:
You changed the planner, but are testing the executor.